trustpinning/trustpin_test.go

```diff
@@ -0,0 +1,33 @@
+package trustpinning
+
+import (
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestWildcardMatch(t *testing.T) {
+	testCerts := map[string][]string{
+		"docker.io/library/ubuntu": {"abc"},
+		"docker.io/endophage/b*":   {"def"},
+		"docker.io/endophage/*":    {"xyz"},
+	}
+
+	// wildcardMatch should ONLY match wildcarded names even if a specific
+	// match is present
+	res, ok := wildcardMatch("docker.io/library/ubuntu", testCerts)
+	require.Nil(t, res)
+	require.False(t, ok)
+
+	// wildcard match should match on segment boundaries
+	res, ok = wildcardMatch("docker.io/endophage/foo", testCerts)
+	require.Len(t, res, 1)
+	require.Equal(t, "xyz", res[0])
+	require.True(t, ok)
+
+	// wildcardMatch should also match between segment boundaries, and take
+	// the first match is finds as the ONLY match.
```

Comment:
Is `docker.io/endophage/b*` actually the first match it finds?  Since we can't depend on `map` order, couldn't the keys have come in any order?  It chooses the longest matching one, right?